lib/adBlockRustUtils.js

```diff
@@ -126,11 +126,56 @@ const generateResourcesFile = async (outLocation) => {
   return fs.writeFile(outLocation, await generateResources(), 'utf8')
 }
 
+/**
+ * A list of requests that should not be blocked unless the list has some serious issue.
+ *
+ * Each element is [requestUrl, sourceUrl, requestType].
+ */
+const sampleUnblockedNetworkRequests = [
+  // real resources from a personal website which will never ship ads/trackers
+  ['https://antonok.com', 'https://antonok.com', 'document'],
+  ['https://antonok.com/style.css?h=9aba43f4dd864e1e4f3a', 'https://antonok.com', 'stylesheet'],
+  ['https://antonok.com/res/font/icons.woff2?h=598bc5b2aa7cdaf390d9', 'https://antonok.com', 'font'],
+  ['https://antonok.com/res/antonok-logo.svg', 'https://antonok.com', 'image'],
+  ['https://antonok.com/processed_images/profile-2021-05-03.14f955fe3ab1a230.webp', 'https://antonok.com', 'image'],
+  // real resources from brave.com
+  ['https://brave.com', 'https://brave.com', 'document'],
+  ['https://brave.com/js/navigation.js', 'https://brave.com', 'script'],
+  ['https://brave.com/static-assets/css-old/main.min.css', 'https://brave.com', 'stylesheet'],
+  ['https://brave.com/static-assets/images/brave-logo-sans-text.svg', 'https://brave.com', 'image'],
+  // real resources from wikipedia.org
+  ['https://en.wikipedia.org/wiki/Ad_blocking', 'https://en.wikipedia.org/wiki/Ad_blocking', 'document'],
+  ['https://en.wikipedia.org/w/load.php?lang=en&modules=codex-search-styles%7Cext.cite.styles%7Cext.pygments%2CwikimediaBadges%7Cext.uls.interlanguage%7Cext.visualEditor.desktopArticleTarget.noscript%7Cskins.vector.icons%2Cstyles%7Cskins.vector.zebra.styles%7Cwikibase.client.init&only=styles&skin=vector-2022', 'https://en.wikipedia.org/wiki/Ad_blocking', 'stylesheet'],
+  ['https://en.wikipedia.org/w/load.php?lang=en&modules=startup&only=scripts&raw=1&skin=vector-202', 'https://en.wikipedia.org/wiki/Ad_blocking', 'script'],
+  ['https://en.wikipedia.org/static/images/mobile/copyright/wikipedia-wordmark-en.svg', 'https://en.wikipedia.org/wiki/Ad_blocking', 'image'],
+  ['https://en.wikipedia.org/static/images/icons/wikipedia.png', 'https://en.wikipedia.org/wiki/Ad_blocking', 'image'],
+  // hypothetical embeddings on example.com
+  ['https://antonok.com', 'https://example.com', 'subdocument'],
+  ['https://brave.com', 'https://example.com', 'subdocument'],
+  ['https://en.wikipedia.org/wiki/Ad_blocking', 'https://example.com', 'subdocument']
+]
+
+/*
+ * Throw an error if the list is blocking any of the resources from `sampleNetworkRequests`.
+ */
+const sanityCheckList = ({ title, format, data }) => {
+  const filterSet = new FilterSet()
+  filterSet.addFilters(data.split('\n'), { format })
+  const engine = new Engine(filterSet)
+  for (const request of sampleUnblockedNetworkRequests) {
+    const result = engine.check(request[0], request[1], request[2])
+    if (result) {
+      throw new Error(title + ' failed sanity check for ' + request + '. Check for corrupted list contents.')
```

Comment:
Where is this Error caught? Is it on line 119, and is that where it's sent to Sentry?

Reply:
Yep, line 119 of `scripts/generateAdBlockRustDataFiles.js`. Same place we already capture warnings about network issues when lists fail to download.

I've also opened https://github.com/brave/devops/issues/11130 to make sure those get reported in Slack going forwards, since they aren't currently.